pkg/asset/manifests/dns.go

```diff
@@ -67,6 +67,8 @@ func (*DNS) Dependencies() []asset.Asset {
 }
 
 // Generate generates the DNS config and its CRD.
+//
+//nolint:gocyclo
```

Comment:
Nit: it's nice to add a reason why it's ok to silence the linter.

Reply:
The way to fix this linting issue is to move the code into its own function, especially if you make the changes to check the `PrivateVisabilityConfig` of the zone